sympy/core/add.py

```diff
@@ -1004,11 +1005,11 @@ def _eval_as_leading_term(self, x, cdir=0):
         except TypeError:
             return expr
 
-        new_expr=new_expr.together()
+        new_expr = new_expr.together().factor()
         if new_expr.is_Add:
             new_expr = new_expr.simplify()
 
-        if not new_expr:
+        if new_expr.is_zero:
```

Comment:
Could `if new_expr.is_zero is not False:` be used?

Reply:
It causes `RecursionError` with -
```py
def test_issue_12769():                                                         
    r, z, x = symbols('r z x', real=True)                                       
    a, b, s0, K, F0, s, T = symbols('a b s0 K F0 s T', positive=True, real=True)
    fx = (F0**b*K**b*r*s0 - sqrt((F0**2*K**(2*b)*a**2*(b - 1) + \               
        F0**(2*b)*K**2*a**2*(b - 1) + F0**(2*b)*K**(2*b)*s0**2*(b - 1)*(b**2 - 2*b + 1) - \
        2*F0**(2*b)*K**(b + 1)*a*r*s0*(b**2 - 2*b +  1) + \                     
        2*F0**(b + 1)*K**(2*b)*a*r*s0*(b**2 - 2*b + 1) - \                      
        2*F0**(b + 1)*K**(b + 1)*a**2*(b - 1))/((b - 1)*(b**2 - 2*b + 1))))*(b*r -  b - r + 1)
                                                                                
    assert fx.subs(K, F0).factor(deep=True) == limit(fx, K, F0).factor(deep=True)
``` 
(particularly the limit part)
with `new_expr = F0**(2*b)*s0*(r - 1)`